paperqa/types.py

```diff
@@ -40,38 +41,39 @@
 logger = logging.getLogger(__name__)
 
 # A context var that will be unique to threads/processes
-cvar_answer_id = contextvars.ContextVar[UUID | None]("answer_id", default=None)
+cvar_session_id = contextvars.ContextVar[UUID | None]("session_id", default=None)
 
 
 @contextmanager
-def set_llm_answer_ids(answer_id: UUID):
-    token = cvar_answer_id.set(answer_id)
+def set_llm_session_ids(session_id: UUID):
+    token = cvar_session_id.set(session_id)
     try:
         yield
     finally:
-        cvar_answer_id.reset(token)
+        cvar_session_id.reset(token)
 
 
 class LLMResult(BaseModel):
     """A class to hold the result of a LLM completion.
 
-    To associate a group of LLMResults, you can use the `set_llm_answer_ids` context manager:
+    To associate a group of LLMResults, you can use the `set_llm_session_ids` context manager:
 
     ```python
-    my_answer_id = uuid4()
-    with set_llm_answer_ids(my_answer_id):
+    my_session_id = uuid4()
+    with set_llm_session_ids(my_session_id):
         # code that generates LLMResults
         pass
     ```
 
-    and all the LLMResults generated within the context will have the same `answer_id`.
+    and all the LLMResults generated within the context will have the same `session_id`.
     This can be combined with LLMModels `llm_result_callback` to store all LLMResults.
     """
 
     id: UUID = Field(default_factory=uuid4)
-    answer_id: UUID | None = Field(
-        default_factory=cvar_answer_id.get,
+    session_id: UUID | None = Field(
+        default_factory=cvar_session_id.get,
         description="A persistent ID to associate a group of LLMResults",
+        alias="answer_id",
     )
```

Comment:
@mskarlin not sure how much this will change things on server side

Reply:
You'll need to opt into Pydantic `ModelConfig(populate_by_name=True)` for `alias` to work for assignment at construction time

Reply:
I think since we overwrite this w. the query id we should be OK

Reply:
@jamesbraza - do I need to do that? I think users can only view them, users should not be constructing them, and I changed the init code.

Reply:
Oh okay sounds like you're aware of this, then feel free to ignore it

Reply:
@jamesbraza - on reflection, I do need to do this though, otherwise we cannot load objects from DB that use `answer` as the field name